### DISCUSSION TITLE
HLG HDR Labs Queries

### DISCUSSION THREAD

**Discussion by RikshaDriver** [gopro/labs#1065] [General]
Just playing with Labs for the H13 and wondering what QR code would be required to explicitly enable/disable HLG.

I see there are also two "Color" settings: Vibrant and Natural. What's the difference in the context of HLG and why do they exist for HLG mode?

Regarding White Balance, I see all the same Kelvin values, Native and Auto are available. Do these explicitly map to Rec.2020 gamut or sensor native gamut?

What QR code would one use to get a standards compliant HLG profile?

**Comment by dnewman-gpsw** [comment]
This is still missing as a QR Code.  Will add to the list.

HLG is Rec2020, but white balance Native is still sensor gamut.

Vibrant and Natural also in Rec709 modes.  These are just GoPro flavors with Rec709 or Rec2020.

**Reply by RikshaDriver** [reply]
Thanks David.

Which "color" would be true to ITU-R BT.2100?

**Reply by dnewman-gpsw** [reply]
There is likely no true mode on any camera. GoPro HDR - HLG is designed to target ITU-R BT.2100  All cameras add some character, so I expect GoPro is too.  Natural is more true than Vibrant.

**Reply by RikshaDriver** [reply]
In my testing of HLG over the years, most camera manufacturers seem to follow the BT.2100 HLG specifications relatively well and many even align their implementations with the BT.2408 recommendations (18% gray @ 38% IRE).

There may be slight deviations in the colors due to different sensors and color matrices, but if they deviate anywhere else, it is called out.

For example Sony HLG1, HLG2 and HLG3 as distinct profiles from standard HLG.

Some recent Lumix cameras also offer HLG with the full data range as a unique flavor alongside the BT.2100 standard compliant HLG (video legal ranges).


On that note, I noticed in the HDR clip metadata from HERO13 that HLG encodings are using full data range and not following video legal range as per the standard. Is this deliberate or a bug?

**Reply by dnewman-gpsw** [reply]
Can you point to where HLG is not supporting full range. Might be a bug, it might be just a less common option.  P.S. Now if this relates to Labs, HLG is not modified by Labs (it least not yet.)

**Reply by RikshaDriver** [reply]
https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.2100-2-201807-I!!PDF-E.pdf

Table 9  + preceding paragraph starting from bottom of page 9.

**Reply by RikshaDriver** [reply]
Would be good to allow for Anamorphic and other Lens mods to work via Labs.

Would it be possible via Labs to have HLG without DOL-HDR to enable >30p?

**Reply by dnewman-gpsw** [reply]
> Would be good to allow for Anamorphic and other Lens mods to work via Labs.

There do now.  oXx commands 0 through 9. Anamorphic will be the command ```oX5``` when that lens is released.

> Would it be possible via Labs to have HLG without DOL-HDR to enable >30p?

Not likely via Labs, but it is reasonable request. I have passed it on.

**Reply by RikshaDriver** [reply]
> > Would be good to allow for Anamorphic and other Lens mods to work via Labs.
> 
> 
> 
> There do now.  oXx commands 0 through 9. Anamorphic will be the command ```oX5``` when that lens is released.
>


Sorry typo meant current HLG with anamorphic and other lens mods



> > Would it be possible via Labs to have HLG without DOL-HDR to enable >30p?
> 
> 
> 
> Not likely via Labs, but it is reasonable request. I have passed it on.
>

Thanks

**Reply by dnewman-gpsw** [reply]
If that happens it will be in main stream firmware, Labs get that stuff for free.

**Comment by tvtechdir** [comment]
Piggy-backing off this thread to keep everything in one spot. Will Labs support any manual exposure settings such as locking the ISO in HLG or being able to adjust the high and low end?

**Reply by RikshaDriver** [reply]
This is a fair ask. I was hoping we would see some more controls for HLG via Labs at least.

Currently it's maxed out to 1600 ISO with no Exposure control options. Not even Spot metering.

Would be good to see some of this functionality added for HLG mode

**Reply by tvtechdir** [reply]
Just re-bumping the thread in case GoPro can add some kind of exposure lock in HLG...or even some kind of setting to slow down the auto-exposure reaction time. Would be extremely nice to have.

**Comment by RikshaDriver** [comment]
Bumping this topic again to see if we will be getting additional controls for HLG mode. There is currently no Spot Metering, and EV Compensation is locked.

Would be good to get some of the Spot Metering, Exposure Adjustment & ISO adjustment features enabled via Labs.

I've found that the current HLG implementation on HERO13 over-exposes by approximately ~1.5 to 1.67 stops relative to HLG BT.2408... i.e. Signal peaks at ~2.5 Stops above Mid-Gray. This is no better than Rec.709. It also likely forces ISO to be cranked up in some scenarios when not required.

Waiting for Mission 1 to arrive so I can test that out, but I'm presuming this is either a deliberate choice, or a bug in the HLG implementation on HERO13?
